strawberryfields/ops.py

```diff
@@ -2222,6 +2222,44 @@ def __str__(self):
         return temp
 
 
+class Ggate(Gate):
+    r"""General N-mode Gaussian gate.
+
+    The general N-mode Gaussian gate can be parametrized by a symplectic matrix S and a displacement parameter d.
+
+    .. math::
+        G(S,d)
+
+    Args:
+
+        S (array): symplectic matrix
+        d (array): vector of displacement parameters
+
+    .. details::
+
+        .. admonition:: Definition
+            :class: defn
+
+        .. math::
+            & cov = S*cov*S^T,\\
+            & mean = mean + d.
+
+    """
+    ns = None
+
+    def __init__(self, S, d):
+        super().__init__([S, d])
+        self.ns = len(d)
+
+    def _apply(self, reg, backend, **kwargs):
+        S, d = par_evaluate(self.p)
+        backend.gaussian_gate(S, d, *reg)
+
+    def _decompose(self, reg, **kwargs):
+        # TODO: decompose?
```

Comment:
@nquesada @josh146 what do you think? We do know how to decompose the Gaussian unitary, but what happens if we tell SF how to do it? Will it try to decompose it all the time and we will lose the advantage of doing it in a single go, or will it decompose it only when we try to compile the circuit?

Reply:
The decomposition will only occur if not supported by the backend :) This is determined by the `sf.compilers` module